components/next_steps_view/steps/enter_support_email/enter_support_email.tsx

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useState} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import Input from 'components/input';
+
+import './enter_support_email.scss';
+import {StepComponentProps} from 'components/next_steps_view/steps';
+import {AdminConfig} from 'mattermost-redux/types/config';
+import {ActionFunc} from 'mattermost-redux/types/actions';
+
+const reg = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+
+type Props = StepComponentProps & {
+    supportEmail: string;
+    actions: {
+        patchConfig: (config: AdminConfig) => ActionFunc;
+    };
+};
+
+function EnterSupportEmail(props: Props): JSX.Element {
+    const [supportEmail, setSupportEmail] = useState('');
+    const [validationError, setValidationError] = useState('');
+
+    useEffect(() => {
+        setSupportEmail(props.supportEmail);
+    }, [props.supportEmail]);
+
+    const updateState = (setStateFunc: (value: string) => void) => {
+        return (event: React.ChangeEvent<HTMLInputElement>) => {
+            setStateFunc(event.target.value);
+        };
+    };
```

Comment:
Curious why wrap this into a function that receives the setStateFunc and not the event handler directly?

Reply:
Just trying to make it more generic. But since we're updating one input field I have removed the wrapper.